terraform/modules/compute/wls_compute/wls_variables.tf

```diff
@@ -14,9 +14,10 @@ variable "wls_edition" {
 variable "wls_admin_user" {
   type        = string
   description = "The name of the admin user that will be added to the WebLogic domain"
+  default     = "wls_user"
   validation {
-    condition     = replace(var.wls_admin_user, "/^[a-zA-Z][a-zA-Z0-9_-]{7,127}/", "0") == "0"
-    error_message = "WLSC-ERROR: The value for wls_admin_user should be between 8 and 128 characters long and alphanumeric, and can contain underscore (_) and hyphen(-) special characters."
+    condition     = replace(var.wls_admin_user, "/^[a-zA-Z][a-zA-Z0-9_-]{7,127}/", "0") == "0" && !contains(["system", "admin", "administrator", "weblogic"], var.wls_admin_user)
+    error_message = "WLSC-ERROR: The value for wls_admin_user should be between 8 and 128 characters long and alphanumeric, and can contain underscore (_) and hyphen(-) special characters, and should not be system, admin, administrator, or weblogic."
   }
 }
```

Comment:
Did you test Production mode(non-secure mode)? Doesnt it block "administrator", "weblogic" for non-secure mode also?